Gui update

M-B-K-0 opened the pull request:
Implemented GUI, business logic and first end to end tests

Commit:
updated UI, will fail the existing UI tests

Commit:
updated gui for floors

Commit:
serviced floors moved to elevator added to gui and mirroring bug in gui fixed

Commit:
added functionality that the button press should now update the gui, it does not...

Commit:
now gui is working correctly on click of the manual button

Commit:
updated gui and added doxygen comments

Commit:
added static gui tests

M-B-K-0 committed:
changed screen size

Commit:
removed unnecessary initializer class and minor cleanup and fixes

Commit:
Merge branch 'gui_update' of github.com:fhhagenberg-sqe-esd-ws22/elevator-control-center-team-b into gui_update

Commit:
updated tests for viewModel so they no longer use sleep

Commit:
Changed the AppTest, so the data of the service is changed in the test cases and not statically defined

Commit:
Merge branch 'gui_update' of https://github.com/fhhagenberg-sqe-esd-ws22/elevator-control-center-team-b into gui_update

Commit:
Fixed the bugs, so the tests are successful

Commit:
implemented control logic and unit tests for it

Commit:
Merge branch 'gui_update' of github.com:fhhagenberg-sqe-esd-ws22/elevator-control-center-team-b into gui_update

Commit:
tried to link change listener to integer property

Commit:
fixed listener

Commit:
gui update button presses in gui linked to business logic

Commit:
added update logic for direction in BusinessLogic,
added setting of the remote in the update function

Commit:
updated ElevatorUpdater with setters and proper unit tests

Commit:
testcases fixed for target and direction updates

Commit:
gui test with robot

Commit:
updated and extended gui test

Commit:
added opacity change to highlight currently pressed button

Commit:
added opacity tests

M-B-K-0 committed:
updated serverargs for xvfb

M-B-K-0 committed:
updated server-args for xvfb

M-B-K-0 committed:
Update github-actions.yml

tried to fix error in automated build by installing at-spi2-core

M-B-K-0 committed:
removed unnecessary changes

Commit:
removed tests that would no longer function due to ElevatorUpdater update

Commit:
Merge branch 'gui_update' of github.com:fhhagenberg-sqe-esd-ws22/elevator-control-center-team-b into gui_update

pholzer commented on src/main/java/at/fhhagenberg/app/ECCApp.java:
```diff
@@ -1,26 +1,68 @@
+        List<FloorUpdater> floorUpdaters = new ArrayList<>();
+        List<ElevatorUpdater> elevatorUpdaters = new ArrayList<>();
+
+        for (Elevator elevator : building.getElevators()) {
```
Warum nimmst do nd de UpdaterFactory? Hod des an Grund oder afoch vagessn?

pholzer commented on src/main/java/at/fhhagenberg/model/Building.java:
```diff
@@ -1,36 +1,27 @@
      * @param floors List of Floor objects which represent all floors in the building
      */
-    public Building(List<Elevator> elevators, List<Floor> floors) {
+    public Building(ArrayList<Elevator> elevators, ArrayList<Floor> floors) {
```
Warum hosd des auf ah ArrayList geändert? des verursocht CodeSmells in so Schnittstöhn soidma nur Interface types verwenden

Commit:
Removed code smells

Commit:
forgot a smell

Commit:
Added mutation tests and added tests to cover mutations where it was possible. Mutation score of 90% was set as requirement for test success.
Added a logger and exception handling.
Exceptions are handled in BuildingViewModel.update and ECCApp.createScene.

Commit:
Attempt to pass quality gate by using a named logger

Commit:
Attempt to pass quality gate by using a named logger

Commit:
Merge branch 'gui_update' of https://github.com/fhhagenberg-sqe-esd-ws22/elevator-control-center-team-b into gui_update

Commit:
Attempt to remove code smells

Commit:
Replaced FileHandler of the Logger with a ConsoleHandler, so quality gate does not fail
Removed code smells

Commit:
explicitly defined max log file size and added a simple formatter.

Commit:
Added a test for Logging.

Commit:
non working elevator rmi lookup added

Commit:
update to enable rmi

Commit:
Changed Logging Library to Logback
Changed getNearestFloor to use getElevatorFloor

Commit:
Merge branch 'gui_update' of https://github.com/fhhagenberg-sqe-esd-ws22/elevator-control-center-team-b into gui_update
